rust/lance-index/src/vector/hnsw.rs

```diff
@@ -279,7 +280,7 @@ impl HNSW {
                 }
             })?)?;
 
-        let mut levels = vec![];
+        let mut levels = Vec::with_capacity(hnsw_metadata.level_offsets.len() - 1);
```

Comment:
nit: if you use an iterator, most has `size_hint`, this can be done automatically.